src/service.js

```diff
@@ -0,0 +1,316 @@
+if(!global._babelPolyfill) { require('babel-polyfill'); }
+
+import Proto from 'uberproto';
+import mongoose from 'mongoose';
+import filter from 'feathers-query-filters';
+import errors from 'feathers-errors';
+import errorHandler from './error-handler';
+
+// Use native promises
+mongoose.Promise = global.Promise;
+
+const Schema = mongoose.Schema;
```

Comment:
I think that would be fine to force the user to pass in an actual mongoose model and then add hooks using app.service('todos').before vs passing them in as part of the model. Not sure what the consensus is among the rest of the group.